components/secret-mgt/org.wso2.carbon.identity.secret.mgt.core/src/test/java/org/wso2/carbon/identity/secret/mgt/core/SecretManagerTest.java

```diff
@@ -466,11 +557,80 @@ private void mockIdentityTenantUtility() {
 
     private void encryptSecret(String secret) throws org.wso2.carbon.core.util.CryptoException {
 
-        when(cryptoUtil.encryptAndBase64Encode(secret.getBytes(Charsets.UTF_8))).thenReturn(ENCRYPTED_VALUE1);
+        if (SAMPLE_SECRET_VALUE2.equals(secret)) {
+            when(cryptoUtil.encryptAndBase64Encode(secret.getBytes(Charsets.UTF_8))).thenReturn(ENCRYPTED_VALUE2);
+        } else {
+            when(cryptoUtil.encryptAndBase64Encode(secret.getBytes(Charsets.UTF_8))).thenReturn(ENCRYPTED_VALUE1);
+        }
     }
 
     private void decryptSecret(String cipherText) throws org.wso2.carbon.core.util.CryptoException {
 
-        when(cryptoUtil.base64DecodeAndDecrypt(cipherText)).thenReturn(SAMPLE_SECRET_VALUE1.getBytes());
+        if (ENCRYPTED_VALUE2.equals(cipherText)) {
+            when(cryptoUtil.base64DecodeAndDecrypt(cipherText)).thenReturn(SAMPLE_SECRET_VALUE2.getBytes());
+        } else {
+            when(cryptoUtil.base64DecodeAndDecrypt(cipherText)).thenReturn(SAMPLE_SECRET_VALUE1.getBytes());
+        }
+    }
+
+    private IdentityProvider buildIDPObject() {
+
+        IdentityProvider idp = new IdentityProvider();
+        idp.setId("5");
+        idp.setIdentityProviderName("testIdP1");
+        idp.setEnable(true);
+        idp.setPrimary(true);
+        idp.setFederationHub(true);
+        idp.setCertificate("");
+
+        FederatedAuthenticatorConfig federatedAuthenticatorConfig = new FederatedAuthenticatorConfig();
+        federatedAuthenticatorConfig.setDisplayName("OIDCAuthenticator");
+        federatedAuthenticatorConfig.setName("SampleOIDCAuthenticator");
+        federatedAuthenticatorConfig.setEnabled(true);
+        Property property1 = new Property();
+        property1.setName(SAMPLE_SECRET_NAME1);
+        property1.setValue(SAMPLE_SECRET_VALUE1);
+        property1.setConfidential(true);
+        Property property2 = new Property();
+        property2.setName("NonSecretProperty");
+        property2.setValue("PropertyValue");
+        property2.setConfidential(false);
+        federatedAuthenticatorConfig.setProperties(new Property[]{property1, property2});
+        idp.setFederatedAuthenticatorConfigs(new FederatedAuthenticatorConfig[]{federatedAuthenticatorConfig});
+
+        return idp;
+    }
+
+    private IdentityProvider buildUpdatedIdpObject(IdentityProvider identityProvider) {
+
+        for (Property property : identityProvider.getFederatedAuthenticatorConfigs()[0].getProperties()) {
+            if (property.isConfidential()) {
+                property.setValue(SAMPLE_SECRET_VALUE2);
+            }
+        }
+
+        return identityProvider;
+    }
+
+    private List<String> idpSecretReferences(IdentityProvider identityProvider) {
```

Comment:
Adhere to the method naming conventions and make this method readable

Reply:
Resolved with [2b47e12](https://github.com/wso2/carbon-identity-framework/pull/4430/commits/2b47e121b1c14f3ddce774f6b6a867c957fb5848)